tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -120,59 +124,121 @@ private async void GetIcons()
         }
     }
 
-    internal async Task<bool> Invoke(int? pid, HWND? hwnd)
+    internal async Task<Process?> Invoke(int? pid, HWND? hwnd)
     {
-        var result = false;
+        var process = default(Process);
 
-        try
+        var parsedArguments = string.Empty;
+        if (!string.IsNullOrEmpty(Arguments))
         {
-            var parsedArguments = string.Empty;
-            if (!string.IsNullOrEmpty(Arguments))
+            var argumentVariables = new Dictionary<string, int>();
+            if (pid.HasValue)
             {
-                var argumentVariables = new Dictionary<string, int>();
-                if (pid.HasValue)
-                {
-                    argumentVariables.Add("pid", pid.Value);
-                }
-
-                if (hwnd.HasValue)
-                {
-                    argumentVariables.Add("hwnd", (int)hwnd.Value);
-                }
+                argumentVariables.Add("pid", pid.Value);
+            }
 
-                parsedArguments = ReplaceKnownVariables(Arguments, argumentVariables);
+            if (hwnd.HasValue)
+            {
+                argumentVariables.Add("hwnd", (int)hwnd.Value);
             }
 
+            parsedArguments = ReplaceKnownVariables(Arguments, argumentVariables);
+        }
+
+        try
+        {
             if (ActivationType == ToolActivationType.Protocol)
             {
-                result = await Launcher.LaunchUriAsync(new Uri($"{parsedArguments}"));
+                // Docs say this returns true if the default app for the URI scheme was launched;
+                // false otherwise. However, if there's no registered app for the protocol, it shows
+                // the "get an app from the store" dialog, and returns true. So we can't rely on the
+                // return value to know if the tool was actually launched.
+                var result = await Launcher.LaunchUriAsync(new Uri($"{parsedArguments}"));

```

Comment:
does new Uri(parsedArguments) work? Seems like a needless roundtrip written as is.

Reply:
Tim - Yes, leftover from when it had more arguments. Fixed now, thanks.